tcms/testplans/views.py

```diff
@@ -335,13 +336,12 @@ def ajax_response(request, queryset, column_names, template_name):
     data = data_table_result.get_response_data()
     data['querySet'] = calculate_stats_for_testplans(data['querySet'])
 
-    # todo: prepare the JSON with the response, consider using :
-    # from django.template.defaultfilters import escapejs
-    json_result = render_to_string(
+    json_result = ast.literal_eval(render_to_string(
         template_name,
         data,
-        request=request)
-    return HttpResponse(json_result, content_type='application/json')
+        request=request))
+
+    return JsonResponse(json_result)
```

Comment:
This entire method is an abomination. It uses a hard-coded template and intermixes that with some values to produce a supposedly valid JSON string. The proper way for this to work is to refactor the entire method and return `JsonResponse(object)` and let the calling UI code to deal with the actual rendering and combining data with the HTML tags.

Drop this change and work on it separately if you will. Warning: this will be quite a big and profound change.